src/http/apis/handle-response-error.ts

```diff
@@ -0,0 +1,18 @@
+import { ResponseError } from "../../errors/request-error";
+import { VehicleUnavailableError } from "../../errors/vehicle-unavailable-error";
+
+const throwErrorByType = (content: any) => {
+    switch (content.type) {
+        case "VehicleUnavailableException":
```

Comment:
I'm not a fan of using backend exception types as dependencies to resolve errors, mainly because I really don't think we should expose exceptions from the backend - if we can control the output. They shouldn't be used as a "control flow", IMO.

If we need to differentiate between errors from backend, we should rely on status codes + possibly error codes/messages included in the response body.

Reply:
Agreed. The status code is 409, but will return an error code as well and skip the exception.